### PR TITLE
Initial Relase Readiness feature

### PR DESCRIPTION
**Pull request by jlarfors**
Sorry for the gigantic PR. A lot of the changes are "ninja" fixes, or aesthetics... things that should be done and with more time would be done in separate PRs.

Main things:
1. Remove `ResourceBlockJSON` as it was a PITA and the `ResourceBlock` can now be marshalled/unmarshalled directly into JSON
2. Added a `builtin` package with a builtin schema, written in HCL, that uses `go generate` to produce Go structs and vars for the schema table
3. Add some more things to the CLI: `release` and `query`

### CODE REVIEW

File: cmd/query/query.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	_       cmdutil.Options = (*QueryOptions)(nil)
+	getLong                 = normalise.LongDesc(`
+		Display one or many bubbly resources
+
+		    $ bubbly get (KIND | ID | all) [flags]
+
+		`)
+
+	getExample = normalise.Examples(`
+		# Display all bubbly resources stored on the bubbly server
```
Comment (jlarfors) [inline cmd/query/query.go:27]: TODO

File: api/common/resource.go [hunk 2]
```diff
@@ -27,6 +27,8 @@ func RunResource(bCtx *env.BubblyContext, ctx *core.ResourceContext, id string,
 		}
 	}
 	runCtx := core.NewResourceContext(inputs, ctx.NewResource, ctx.Auth)
+	// TODO: handle this better... it get's copied around in multiple places...
```
Comment (jlarfors) [inline api/common/resource.go:30]: TODO

File: api/v1/pipeline.go [hunk 3]
```diff
@@ -50,9 +50,10 @@ func (p *Pipeline) Apply(bCtx *env.BubblyContext, ctx *core.ResourceContext) cor
 		runCtx := core.NewResourceContext(
 			ctx.State.Value([]string{"task"}, ctx.Inputs), ctx.NewResource, ctx.Auth,
 		)
+		// TODO: how to handle this better?!
```
Comment (jlarfors) [inline api/v1/pipeline.go:53]: TODO

File: bubbly/release_eval.go [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+)
+
+func EvalReleaseCriteria(bCtx *env.BubblyContext, criteriaName string) (*ReleaseSpec, error) {
+	var fileParser BubblyFileParser
+	err := parser.ParseConfig(bCtx, &fileParser)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing bubbly configs: %w", err)
+	}
+
+	// TODO: hacked/hardcoded -- get data from fileParser.Relase
```
Comment (jlarfors) [inline bubbly/release_eval.go:23]: TODO

File: bubbly/builtin/gen/gen.go [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+		for _, field := range table.Fields {
+			fmt.Fprintf(&b, "\t\t\tfield(\"%s\", %s, %t),\n", field.Name, field.Type.GoString(), field.Unique)
+		}
+		fmt.Fprintf(&b, "\t\t),\n")
+		fmt.Fprintf(&b, "\t\tjoins(\n")
+		for _, join := range table.Joins {
+			fmt.Fprintf(&b, "\t\t\tjoin(\"%s\", %t, %t),\n", join.Table, join.Single, join.Unique)
+		}
+		fmt.Fprintf(&b, "\t\t),\n")
+		fmt.Fprintf(&b, "\t),\n")
```
Comment (olliefr) [inline bubbly/builtin/gen/gen.go:56]: Would [text/template](https://golang.org/pkg/text/template/) not help with this kind of thing, @jlarfors?

Comment (jlarfors) [reply bubbly/builtin/gen/gen.go:56]: I did initially start using it, but then I was being less productive so I just moved to `Fprintf`. If this get's more complex/requires constant changing then maybe `text/template` could be introduced

**Comment by jlarfors** [comment]
Closing as opened up new branch with this work

**Pull request closed without merging**